modules/core/arrow-test/src/main/kotlin/arrow/test/laws/SemialignLaws.kt

```diff
@@ -0,0 +1,170 @@
+package arrow.test.laws
+
+import arrow.Kind
+import arrow.core.Const
+import arrow.core.Either
+import arrow.core.Ior
+import arrow.core.None
+import arrow.core.Option
+import arrow.core.Some
+import arrow.core.Tuple2
+import arrow.core.extensions.eq
+import arrow.core.extensions.ior.eq.eq
+import arrow.core.extensions.list.functorFilter.flattenOption
+import arrow.core.extensions.list.monadFilter.filterMap
+import arrow.core.toT
+import arrow.typeclasses.Eq
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Semialign
+import io.kotlintest.properties.Gen
+import io.kotlintest.properties.forAll
+
+object SemialignLaws {
+
+  val iorEq1 = Ior.eq(Int.eq(), Int.eq())
+  val iorEq2 = Ior.eq(Int.eq(), iorEq1)
+
+  fun <F> laws(
+    SA: Semialign<F>,
+    gen: Gen<Kind<F, Int>>,
+    buildEq: (Eq<*>) -> Eq<Kind<F, *>>,
```

Comment:
Is there a reason not to just force `Eq<Kind<F, Int>>`?

Reply:
my idea here was to let SemialignLaws create the Eq and have one generic function to allow tests to put the Eq in a specific context (e.g. ListK.eq). the function is used with different arguments types (see laws construction in SemialignLaws) so the type Eq<Kind<F, Int>> wouldn't fit in all cases. i also think i should enhance this, not yet sure how.

Reply:
Yeah right, this is actually needed and it's a decent solution. Something better would be lifted equality like eq1 from haskell but that's a different story altogether

Reply:
i looked Eq1 up. As it does not seem to exist in arrow i tried to implement it for Option.
it seems straight forward to do and indeed would enhance the mentioned code as the tests then just would pass in an instance of Eq1.

```
MyLaws.laws(Option.semialign(),
        Gen.option(Gen.int()) as Gen<Kind<ForOption, Int>>,
        Option.eq1() 
      )
```

is there a particular reason why Eq1 does not exist in arrow? it looks quite useful

Reply:
It is quite useful :) Same with the other lifted typeclass versions (Show1, Ord1,...)
It's actual uses are quite rare tho, the ones in the tests here are perfect, but the usual case for types is that the inner type is abstract and the outer is concrete, whereas lifted instances like Eq1 come into use when the outer type is abstract and the inner is concrete.
Can't hurt to have it tho :)